ypathgen/pathgen.go

```diff
@@ -956,7 +960,13 @@ func generateChildConstructorsForList(methodBuf *strings.Builder, listAttr *ygen
 		// Create the string for the method parameter list, docstrings, and ygot.NodePath's key list.
 		fieldData.KeyParamListStr = strings.Join(paramListStrs, ", ")
 		fieldData.KeyParamDocStrs = paramDocStrs
-		fieldData.KeyEntriesStr = strings.Join(keyEntryStrs, ", ")
+		if simplifyWildcardPaths && comboIndex == 0 {
+			// The zeroth index has everything wildcarded, so it doesn't need to specify any key value.
+			// This is a tweak that helps with vendor compatibility.
```

Comment:
Which vendor compatibility? It's also not just for compatibility but also according to the [path spec](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-path-conventions.md).

Reply:
apparently broadcom, I don't know the details. Technically both are equivalent right? But it looks like the key=* syntax isn't yet supported.

Reply:
I'll make it clear in the comments that it's part of the spec.

Reply:
Improved comment.